core/execution/market.go

```diff
@@ -3166,16 +3207,34 @@ func (m *Market) commandLiquidityAuction(ctx context.Context) {
 func (m *Market) tradingTerminated(ctx context.Context, tt bool) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.mkt.State = types.MarketStateTradingTerminated
-	m.mkt.TradingMode = types.MarketTradingModeNoTrading
-	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
-	m.stateChanged = true
 
-	if m.settlementPriceInMarket == nil {
-		m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+	if m.mkt.State != types.MarketStateProposed && m.mkt.State != types.MarketStatePending {
+		m.mkt.State = types.MarketStateTradingTerminated
+		m.mkt.TradingMode = types.MarketTradingModeNoTrading
+		m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
+		m.stateChanged = true
+
+		if m.settlementPriceInMarket == nil {
+			m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+			return
+		}
+		m.settlementPriceWithLock(ctx)
+	} else {
+		for party := range m.parties {
```

Comment:
why not move this loop into closeCancelledMarket?

Reply:
I dont understand, do you mean to move `CancelAllOrders` step inside the `closeCanceledMarket`?
I think these should be separate steps - the way it is handled for closing market in the normal way.

Reply:
primarily what I'm saying is that the call to closeCancelledMarket should not be done inside this loop.

Reply:
This has been changed here https://github.com/vegaprotocol/vega/pull/5668/commits/a11e922add05a5659da732ef070f6e0cc73cb00f#diff-1782f5982916957ff0e7e82b0c7130c6d4c8ab4eb2e06f5c7b5b4a5925e15bcfR3217
Missed that
Thanks!